src/domain/accounts/counterfactual-safes/entities/counterfactual-safe.entity.ts

```diff
@@ -0,0 +1,18 @@
+import { RowSchema } from '@/datasources/db/entities/row.entity';
+import { AccountSchema } from '@/domain/accounts/entities/account.entity';
+import { AddressSchema } from '@/validation/entities/schemas/address.schema';
+import { z } from 'zod';
+
+export type CounterfactualSafe = z.infer<typeof CounterfactualSafeSchema>;
+
+export const CounterfactualSafeSchema = RowSchema.extend({
+  chain_id: z.string(),
+  creator: AddressSchema,
+  fallback_handler: AddressSchema,
+  owners: z.array(AddressSchema),
+  predicted_address: AddressSchema,
+  salt_nonce: z.string(),
+  singleton_address: AddressSchema,
+  threshold: z.number().int().gte(0),
+  account_id: AccountSchema.shape.id,
+});
```

Comment:
Maybe we should extend the `CreateCounterfactualSafeDtoSchema` or inidividual field schemas if the property names are changed.

Reply:
Do you mean extending DTOs from entity schemas? I haven't considered it, while I think in this case might make sense, I don't see it as a common pattern as DTOs shape does not match its _target entity_ schema necessarily, wdyt? (I can do it for this one, for sure)

Reply:
Let's keep it as tis then.